### ISSUE TITLE
Shell.TabBarIsVisible = "False" not working on WinUI

### ISSUE THREAD

**Issue by Cywizz** [dotnet/maui#7246]
### Description

Setting the Shell.TabBarIsVisible=False has no effect on WinUI platform. It works as expected on Android.
The issue is the same when you set this property on both XAML and the Page's code behind.

Not sure if it is related to issues: #6746 because my issue is on the TabBar, not the individual Tab.

### Steps to Reproduce

1. Create a MAUI App
2. In AppShell.xaml, add a TabBar and a couple of Tabs with their ContentTemplate pages.
3. Create a page that will be navigated to, which is not declared in the AppShell.xaml
4. Add a button to one of the ContentTemplate pages created in step two.
5. Register the route to that page
6. In the clicked event of the button in step 4, manually navigate to the page created in step 3.

You will see the tabbar disappears on Android, but not on Windows.

### Version with bug

Release Candidate 3 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows SDK 10.0.19041

### Did you find any workaround?

No

**Comment by jfversluis** [comment]
Yep I definitely still see some weird behavior here not just with the TabBar not disappearing, but if you go around pages a couple of times the TabBar does disappear but also never comes back.

Repro here: https://github.com/jfversluis/ReproMaui7246

**Comment by XamlTest** [comment]
Verified this issue with Visual Studio Enterprise 17.3.0 Preview 2.0 [32515.61.main]. Repro on Windows with https://github.com/jfversluis/ReproMaui7246 project.

**Comment by xtuzy** [comment]
Click Button SetTabBarIsVisible=False, only after click other Tabbar and Click current Tabbar, will hide tabbar

https://user-images.githubusercontent.com/17793881/177071847-72c889f5-28a4-43b8-a274-ac1a8f4edb0a.mp4

**Comment by ProvidenceSoftwareLimited** [comment]
Has anyone found a solution to this issue?

**Comment by fgiacomelli** [comment]
Same problem for me, a simple binding like this:
    Shell.TabBarIsVisible="{Binding FromRegistration}"
doesn't work when set from queryparameter received from Shell navigation.

Frustrating

**Comment by rogerbriggen** [comment]
@Redth This blocks our navigation because sometimes the user is not allowed to navigate to other main tabs and on Windows it not only does not work but it will then hide the tabbar sometimes later so you cannot navigate at all and you have to kill the app. Any plans on fixing the shell on windows?

**Comment by bronteq** [comment]
This issue is not solved in Maui 8.0.92

It seems that it was solved here https://github.com/dotnet/maui/pull/17069 but then this pr caused a new bug solved by this https://github.com/dotnet/maui/pull/18155 maybe something got lost?
@Foda @emaf

**Comment by jfversluis** [comment]
Should be fixed by #17069
